torch/csrc/jit/tensorexpr/tensorexpr_init.cpp

```diff
@@ -110,22 +110,33 @@ void initTensorExprBindings(PyObject* module) {
   py::class_<BufHandle, ExprHandle>( // NOLINT
       te,
       "BufHandle")
-      .def(py::init<
-           const std::string&,
-           const std::vector<ExprHandle>&,
-           Dtype>());
+      .def(
+          py::init<const std::string&, const std::vector<ExprHandle>&, Dtype>())
+      .def(
+          "load",
+          [](BufHandle& self,
+             const std::vector<ExprHandle>& v) {
+            return Load::make(self, v);
+          });
 
   py::class_<Placeholder>(te, "Placeholder")
       .def(py::init<
            const std::string&,
            const Dtype&,
            std::vector<ExprHandle>&>())
-      .def("load", [](Placeholder& self, const std::vector<ExprHandle>& v) {
-        return self.load(v);
-      });
+      .def(
+          "load",
+          [](Placeholder& self, const std::vector<ExprHandle>& v) {
+            return self.load(v);
+          })
+      .def(
+          "buf",
+          [](Placeholder& self) { return BufHandle(self.data()); },
+          py::return_value_policy::reference);
   py::class_<Tensor, std::unique_ptr<Tensor, py::nodelete>>(te, "Tensor")
-      .def(py::init(
-          [](BufHandle& b, Stmt* s) { return new Tensor(b.node(), s); }))
+      .def(py::init([](BufHandle& b, Stmt* s) {
+        return std::unique_ptr<Tensor, py::nodelete>(new Tensor(b.node(), s));
```

Comment:
We could probably return the raw pointer here? Why do we use unique_ptr?

Reply:
I kind of assumed that if the class uses an `unique_ptr` you needed to return an unique pointer? But it seems like pybind does some type inference so maybe that's not needed.

Reply:
Yeah, I'm not sure on that either :) both options are fine with me!